### PR TITLE
[docs] consistent code blocks in tensor data guide

### PR DESCRIPTION
**Pull request by maxpumperla**
we're mixing too many doctest styles. specifically the `doctest` directive leads to ugly `>>>` everywhere that also get copied. this PR makes ensures consistency in one of the docs I found that broke this pattern.

### CODE REVIEW

File: doc/source/data/working-with-tensors.rst [hunk 1]
```diff
@@ -33,40 +33,51 @@ Batches of fixed-shape tensors
...
     dtype('uint8')
-    >>> batch["image"][0].shape  # doctest: +SKIP
+    batch["image"][0].shape  # doctest: +SKIP
+    batch["image"][3].shape  # doctest: +SKIP
+
+.. testoutput::
+
     (375, 500, 3)
-    >>> batch["image"][3].shape  # doctest: +SKIP
     (333, 465, 3)
```
Comment (bveeramani) [inline doc/source/data/working-with-tensors.rst:81]: This example emphasizes object representations, so I intentionally chose to use doctest-style. I felt it illustrated the API better:

**doctest-style**
image

**code-block-style**
image

File: doc/source/data/working-with-tensors.rst [hunk 2]
```diff
@@ -33,40 +33,51 @@ Batches of fixed-shape tensors
 
 If your tensors have a fixed shape, Ray Data represents batches as regular ndarrays.
 
-.. doctest::
```
Comment (bveeramani) [inline doc/source/data/working-with-tensors.rst:36]: You can't copy the REPL prompts. They're not selectable.

image

https://github.com/ray-project/ray/blob/790a16d4f5bc748cb405cb00c83b901118dda284/doc/source/conf.py#L118-L124

**Comment by maxpumperla** [comment]
@bveeramani I get your point, but don't you think it's practically impossible for our users to deduce this logic? I'd always advocate for clarity and consistency first. It simply looks bad to suddenly mix in ">>>" in a document.

**Comment by ericl** [comment]
My 2c:
- I don't think the `code-block-style` is intuitive. I'd prefer we stop using this style, as it's hard to tell what is the input and what is the output block. It took me a few seconds to visually "parse" the output when I saw this the first time.
- We are also using inline comments to show outputs in other places, which seems pretty readable to me: https://docs.ray.io/en/master/data/consuming-data.html#retrieving-a-limited-set-of-rows

For this PR, I think we should either use inline comments with normal code blocks, or doctest-style.

**Comment by bveeramani** [comment]
> ...it's hard to tell what is the input and what is the output block. It took me a few seconds to visually "parse" the output when I saw this the first time.

Maybe we can add labels and change the color for code and its outputs? I think Torch and Stripe do a good job of this.

image

image

**Comment by bveeramani** [comment]
> * We are also using inline comments to show outputs in other places, which seems pretty readable to me: https://docs.ray.io/en/master/data/consuming-data.html#retrieving-a-limited-set-of-rows

The inline comments aren't tested, and as a result, they're often incorrect. So, we're trying to move towards tested formats like `doctest` or `testoutput`.

**Pull request closed without merging**